bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/GC100IRBindingProvider.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.gc100ir;
+
+import org.openhab.core.binding.BindingProvider;
+
+/**
+ * Binding provider interface. Defines how to get properties from a binding
+ * configuration.
+ * 
+ * @author Parikshit Thakur & Team
+ * @since 1.6.0
+ */
+public interface GC100IRBindingProvider extends BindingProvider {
+	/**
+	 * Gets the Global Cache IR Instance Name associated with a given <code>itemname</code>.
+	 * @param itemname The item name associated with an instance of a Global Cache IR.
+	 * @return String format of the Global Cache Instance Name
+     */
```

Comment:
Please apply the formatter and cleanup rules to all Java files using the new [IDE setup](https://github.com/openhab/openhab-distro/blob/master/docs/sources/development/ide.md).